### PR TITLE
fix(shadowtable): it occurs NPE while creating shadow table comparing tasks

### LINKED ISSUE #2039: [Bug]: Create shadow table synchronization failed
### ODC version

odc424

### OB version

*

### What happened?

image

### What did you expect to happen?

Created successfully

### How can we reproduce it (as minimally and precisely as possible)?

Create shadow table synchronization

### Anything else we need to know?

_No response_

### Cloud

_No response_

### PR DESCRIPTION
**Pull request by MarkPotato777**
#### What type of PR is this?

type-bug

#### What this PR does / why we need it:
It occurs NPE because before the shadow table tasks have been bound to a flow instance, the `ShadowTableComparingTaskEntity#getId` will be null. If this is the case, we just need to compare the task's creatorId and currentUserId.

#### Which issue(s) this PR fixes:

Fixes #2039 

#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

File: server/odc-service/src/main/java/com/oceanbase/odc/service/structurecompare/StructureComparisonService.java [hunk 1]
```diff
@@ -147,6 +148,12 @@ public List<StructureComparisonTaskResultEntity> batchCreateTaskResults(List<DBO
     }
 
     private void checkPermission(@NonNull StructureComparisonTaskEntity taskEntity) {
+        if (taskEntity.getFlowInstanceId() == null) {
```
Comment (PeachThinking) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/structurecompare/StructureComparisonService.java:151]: structure compare task is bind to flow instance id, the FlowInstanceId would not be null

Comment (MarkPotato777) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/structurecompare/StructureComparisonService.java]: okay~ I removed it.

**Review by yhilmare — APPROVED** [review]
lgtm

**Review by PeachThinking — APPROVED** [review]
LGTM

**Pull request merged**